core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -270,6 +270,8 @@ protected Object primaryReturnHandler(InvocationContext ctx, AbstractDataWriteCo
       }
       MapResponseCollector collector = MapResponseCollector.ignoreLeavers(isReplicated, owners.size());
       RpcOptions rpcOptions = rpcManager.getSyncRpcOptions();
+      // Mark the command as a backup write so it can skip some checks
+      command.addFlags(FlagBitSets.BACKUP_WRITE);
```

Comment:
I think you have to remove the flag after sending (after line 280 below). if the command retries (topology changed), the command will have this flag on the primary owner.

Reply:
Makes sense, let me take a closer look.